### ISSUE TITLE
Identify self-tested accessibility fixes

### ISSUE THREAD

**Issue by LucyGwilliamAdmin** [open-sdg/open-sdg#1593]
Indicator pages:
1. [ ] The "Sub-categories" label is an H4, and I don't believe its parents have H3, H2, H1. Lighthouse points to this page: https://web.dev/heading-order/?utm_source=lighthouse&utm_medium=devtools
2. [x] The chart footer items (a `<dl>` element) has an empty `<dl>` inside of it. Lighthouse points to this page: https://web.dev/definition-list/?utm_source=lighthouse&utm_medium=devtools
3. [ ] The chart canvas, the active data tab, and the active metadata tab are giving this: https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#focusable_elements_should_have_interactive_semantics
4. [x] The “Clear selections” does not meet color contrast
5. [ ] The goal icon on indicator pages doesn't have a tab focus outline
6. [ ] When I switched to the iframe tab (as opposed to "Chart" and "Table") the screenreader didn't say anything.
7. [ ] There is no screenreader feedback when a table column is sorted
8. [x] There is no screenreader feedback when data is added using the disaggregation controls (I believe we had that working before, so might be a regression bug)

Site-wide:

1. [ ] Anything using a Font Awesome icon generates and error related to this:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#clickable_elements_must_be_focusable_and_should_have_interactive_semantics. Examples: search bar icon, language toggle icon, disagg clear button icon
Also occurring on the <div id=”indicatorData”> element. These may be a false positive - need to investigate.
2. [x] The “Cookie settings” link is giving this: https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#interactive_elements_must_be_able_to_be_activated_using_a_keyboard (I also noticed that the "Cookie settings" item cannot be tabbed to, so definitely needs fixing)
3. [ ] There is no screenreader feedback when the contrast toggle is used
4. [x] Screenreader misreads Goal 8 as ‘Decent Jobs and Economic Growth’ instead of 'Decent Work and Economic Growth' - PR opened: https://github.com/open-sdg/sdg-translations/pull/277
5. [x] Screenreader misreads Goal 16 as ‘Peace and Justice, Strong Institutions’ instead of 'Peace, Justice and Strong Institutions' - PR opened: https://github.com/open-sdg/sdg-translations/pull/277
6. [x] Goal tiles don't meet colour contrast requirements - this is a known issue and we can't do anything about it other than provide a high contrast version

Reporting Status:

- [x]  The reporting status tab does not have valid values [aria-*]: meaning it cannot be interpreted by the screenreader - see comment below

**Comment by brockfanning** [comment]
Some complaints by Chrome's "Lighthouse report":
1. The "Sub-categories" label is an H4, and I don't believe its parents have H3, H2, H1. Lighthouse points to this page: https://web.dev/heading-order/?utm_source=lighthouse&utm_medium=devtools
2. The chart footer items (a `<dl>` element) has an empty `<dl>` inside of it. Lighthouse points to this page: https://web.dev/definition-list/?utm_source=lighthouse&utm_medium=devtools

Some complaints by Firefox's accessibility scan:

1. Anything using a Font Awesome icon generates and error related to this:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#clickable_elements_must_be_focusable_and_should_have_interactive_semantics. Examples: search bar icon, language toggle icon, disagg clear button icon
Also occurring on the <div id=”indicatorData”> element. These may be a false positive - need to investigate.
2. The chart canvas, the active data tab, and the active metadata tab are giving this: https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#focusable_elements_should_have_interactive_semantics
3. The “Cookie settings” link is giving this: https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#interactive_elements_must_be_able_to_be_activated_using_a_keyboard (I also noticed that the "Cookie settings" item cannot be tabbed to, so definitely needs fixing)
4. The “Clear selections” does not meet color contrast

Other things noticed from manual testing:
1. The goal icon on indicator pages doesn't have a tab focus outline
2. When I switched to the iframe tab (as opposed to "Chart" and "Table") the screenreader didn't say anything.
3. There is no screenreader feedback when a table column is sorted
4. There is no screenreader feedback when the contrast toggle is used
5. There is no screenreader feedback when data is added using the disaggregation controls (I believe we had that working before, so might be a regression bug)

**Comment by brockfanning** [comment]
I added checkboxes above - my thinking is that a checked box means that a separate ticket has been created for it.

**Comment by otis-bath** [comment]
@brockfanning @LucyGwilliamAdmin @phillipgardiner - Some findings from internal testing on the various pages of the UK site. Again same principle applies with the checklist. 

Home:
- [ ] 	Screenreader misreads Goal 8 as ‘Decent Jobs and Economic Growth’ instead of 'Decent Work and Economic Growth'
- [ ]  Screenreader misreads Goal 16 as ‘Peace and Justice, Strong Institutions’ instead of 'Peace, Justice and Strong Institutions'

Goals:
- [ ] Goal tiles don't meet colour contrast requirements

Reporting Status:
- [ ]  The reporting status tab does not have valid values [aria-*]: meaning it cannot be interpreted by the screenreader

Cookies Page:
- [ ]  The cookies radio button fieldset does not have a legend meaning it is passed over completely by the screenreader

User Personas:
- [ ]  Visual data to raw data arrow diagram has incredibly long descriptive text when read aloud. The labels on this image also cannot be viewed in the high contrast mode.

Archived Indicators:
- [ ] Heading level skipped – goes from H1 (Archived indicators) straight to H4 (1.a.1…)

I would say none of these issues are so major they need to hold up the 2.0.0 release but something to consider for 2.1.0

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning re. Reporting status page, so looking at the example on this page (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls), I think I see the issue:

The reporting status tab panel has `id="goalsview"` but the tab has `aria-controls="goals"` - I think it should be `aria-controls="goalsview"`

Similarly, I think the tab for the disaggregation status should have `aria-controls="disaggregationsview"`

Is this right? How long do you think it would take to implement?

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin That sounds right to me. I put up #1687 which should help with that. I tried to do the same for the "extra fields" tabs as well.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning @otis-bath hope you don't mind but I've put a single comment with all of the identified accessibility issues at the top of this ticket just to help streamline - so think we can delete the individual comments?
